### PR TITLE
adding input validation

### PR DESCRIPTION
**Pull request by Guzovatii-L**
please, don't laugh!!

### CODE REVIEW

File: app/event-creator.tsx [hunk 1]
```diff
@@ -1,14 +1,63 @@
 'use client'
 
-import React from "react";
+import React, { useState } from "react";
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
 
 export default function EventCreator() {
 
+
```
Comment (DianaJacota) [inline app/event-creator.tsx:9]: What is the purpose of these empty lines? 🤔

Comment (Guzovatii-L) [reply app/event-creator.tsx:9]: I forgot about it

File: app/event-creator.tsx [hunk 2]
```diff
@@ -1,14 +1,63 @@
...
     const [calendarId, setCalendarId] = React.useState("");
 
+
+    function validateInput() {
+
+        var valid = true;
+
+        if (name.length == 0) {
+            const el = document.getElementById("name");
+            el.style.border = "2px solid red";
```
Comment (DianaJacota) [inline app/event-creator.tsx:22]: this string repeats 3 times in this function. Couldn't we extract it in a variable?

File: app/event-creator.tsx [hunk 3]
```diff
@@ -1,14 +1,63 @@
...
+
+            valid = false;
+            el.onchange = () => {el.style.border = "1px solid grey"};
```
Comment (DianaJacota) [inline app/event-creator.tsx:25]: same mention as above. Could we extract "1px solid grey" in a variable?

Comment (Guzovatii-L) [reply app/event-creator.tsx:25]: maybe

File: app/event-creator.tsx [hunk 4]
```diff
@@ -1,14 +1,63 @@
...
+        }
+
+        if (!valid) {
+
+            alert('Invalid Input')
+            return;
+        }
+
+
+        if (valid)
```
Comment (DianaJacota) [inline app/event-creator.tsx:51]: Even though the brackets are unnecessary when there is a single line, maybe we should have them for consistency. What do you think?

File: app/layout.tsx [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-      <body>{children}</body>
-    </html>
-  )
-}
+    return (
+        <html lang="en">
+            <body>{children}</body>
+        </html>
+    )
+}
```
Comment (DianaJacota) [inline app/layout.tsx:13]: It is a good practice to have an empty line at the end of the file. Without it we have this red sign

File: app/event-creator.tsx (hunk 3 above)
Comment (DianaJacota) [inline app/event-creator.tsx:25]: test

**Review by Guzovatii-L — COMMENTED** [review]
!

**Review by DianaJacota — CHANGES_REQUESTED** [review]
It looks good. Just a few suggestions to address

**Comment by Guzovatii-L** [comment]
Addressed code review suggestions, please check again!!! cc @DianaJacota

File: app/layout.tsx (hunk 5 above)
Comment (aguzovatii) [inline app/layout.tsx:13]:
```suggestion
  return (
    <html lang="en">
      <body>{children}</body>
    </html>
  )
}

```

File: app/layout.tsx [hunk 6]
```diff
@@ -1,13 +1,13 @@
 import './global.css'
 
 export default function RootLayout({
-  children,
+    children,
 }: {
-  children: React.ReactNode
+    children: React.ReactNode
```
Comment (aguzovatii) [inline app/layout.tsx:6]:
```suggestion
  children: React.ReactNode
```

File: app/layout.tsx [hunk 7]
```diff
@@ -1,13 +1,13 @@
 import './global.css'
 
 export default function RootLayout({
-  children,
+    children,
```
Comment (aguzovatii) [inline app/layout.tsx:4]:
```suggestion
  children,
```

**Review by aguzovatii — COMMENTED** [review]
Looks good.
Please make sure to revert all the changes to the files that shouldn't be modified.
I added a few suggestions to revert the unnecessary changes, you can also do that to `page.tsx` file

**Comment by Guzovatii-L** [comment]
Addressed code review suggestions, please check again!!! cc @DianaJacota @aguzovatii

File: app/event-creator.tsx [hunk 8]
```diff
@@ -1,14 +1,57 @@
...
+            valid = false;
+            el.onchange = () => { el.style.border = initstyle, calendarId => setCalendarId(calendarId) };
+        }
+
+        if (!valid) {
+            alert('Invalid Input')
+            return;
+        }
+
+        if (valid) {
```
Comment (DianaJacota) [inline app/event-creator.tsx:48]: Is this if really needed? If the input is invalid you show an alert and return. Could this be written as:
`if(valid) {
    handleClick();
    return:
}

alert('Invalid Input')
`

Comment (Guzovatii-L) [reply app/event-creator.tsx:48]: i know about this, but I left it for clarity :)

**Review by DianaJacota — APPROVED** [review]
Thank you for these changes! It looks great
Congratulations for you first PR🥳🍪

**Pull request merged**